arangod/Pregel/Status/ConductorStatus.h

```diff
@@ -33,34 +34,45 @@
 #include <Cluster/ClusterTypes.h>
 
 #include <Pregel/Common.h>
-#include <Pregel/Status/WorkerStatus.h>
+#include <Pregel/Status/Status.h>
 
 namespace arangodb::pregel {
 
-struct ConductorStatus {
-  TimeStamp timeStamp = std::chrono::system_clock::now();
-  std::unordered_map<ServerID, WorkerStatus> workers;
+struct AccumulatedConductorStatus {
+  Status status;
+  std::unordered_map<ServerID, Status> workers;
+  bool operator==(AccumulatedConductorStatus const&) const = default;
+};
+
+template<typename Inspector>
+auto inspect(Inspector& f, AccumulatedConductorStatus& x) {
+  return f.object(x).fields(
+      f.field("aggregatedStatus",
+              x.status),  // TODO status entries should be on top level
+      f.field("workerStatus", x.workers));
+}
 
+struct ConductorStatus {
+  std::unordered_map<ServerID, Status> workers;
   static auto forWorkers(std::vector<ServerID> const& ids) -> ConductorStatus {
     auto status = ConductorStatus{};
     for (auto const& id : ids) {
-      status.workers.emplace(id, WorkerStatus{});
+      status.workers.emplace(id, Status{});
     }
     return status;
   }
-  auto updateWorkerStatus(ServerID const& id, WorkerStatus const& status)
-      -> void {
-    workers.at(id) = status;
-    timeStamp = status.timeStamp;
+  auto updateWorkerStatus(ServerID const& id, Status&& status) -> void {
+    workers[id] = status;
```

Comment:
Check whether `id` exists in the workers map; at the very least use `at` to get an exception and not UB.